service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -179,23 +178,25 @@ private static boolean processHopByHopHeaders(HttpSession src, HttpSession dest)
         }
 
         // Add source session headers to destination session
-        for(Map.Entry<String, List<String>> e : src.getReadHeaders().entrySet()) {
+        for (Map.Entry<String, List<String>> e : src.getReadHeaders().entrySet()) {
             String name = e.getKey();
-            for(String value : e.getValue()) {
+            for (String value : e.getValue()) {
                 if (!hopByHopHeaders.contains(name)) {
-                    dest.addWriteHeader(name, value);
+                    if (name.equalsIgnoreCase(HttpHeaders.HEADER_HOST)) {
+                        dest.addWriteHeader(name, dest.getRemoteAddress().getResource().getAuthority());
+                    } else {
```

Comment:
Can you also look at HttpConnectProcessor where it adds Host header ? Also see the history w.r.t Host header and may be a fix in that file would be better. Also, I need to think a bit here. 

We may provide a config option to override the Host header. In that case, we need to see what the fix would be.